src/main/java/org/sagebionetworks/bridge/services/ParticipantFileService.java

```diff
@@ -0,0 +1,172 @@
+
+    private ParticipantFileDao participantFileDao;
+
+    private AmazonS3 s3Client;
+
+    private Environment env;
+
+    private String bucketName;
+
+    @Autowired
+    final void setParticipantFileDao(ParticipantFileDao dao) {
+        this.participantFileDao = dao;
+    }
+
+    @Autowired
+    final void setConfig(BridgeConfig config) {
+        bucketName = config.getHostnameWithPostfix("participant-file");
+        env = config.getEnvironment();
+    }
+
+    @Resource(name = "participantFileS3Client")
+    final void setS3client(AmazonS3 s3) {
+        this.s3Client = s3;
+    }
+
+    /**
+     * Get a ForwardCursorPagedResourceList of ParticipantFiles from the given userId, with nextPageOffsetKey set.
+     * If nextPageOffsetKey is null, then the list reached the end and there does not exist next page.
+     *
+     * @param userId the id of the StudyParticipant
+     * @param offsetKey the nextPageOffsetKey.
+     *                  (the exclusive starting offset of the query, if null, then query from the start)
+     * @param pageSize the number of items in the result page
+     * @return a ForwardCursorPagedResourceList of ParticipantFiles
+     */
+    public ForwardCursorPagedResourceList<ParticipantFile> getParticipantFiles(String userId, String offsetKey, int pageSize) {
+        checkNotNull(userId);
+        checkArgument(isNotBlank(userId));
+
+        // pageSize validation is checked in DAO
+        return participantFileDao.getParticipantFiles(userId, offsetKey, pageSize);
+    }
+
+    /**
+     * Returns this ParticipantFile metadata for download. If this file does not exist,
+     * throws EntityNotFoundException.
+     *
+     * @param userId the userId to be queried
+     * @param fileId the fileId of the file
+     * @return the ParticipantFile for download if this file exists
+     * @throws EntityNotFoundException if the file does not exist.
+     */
+    public ParticipantFile getParticipantFile(String userId, String fileId) {
+        checkNotNull(userId);
+        checkNotNull(fileId);
+        checkArgument(isNotBlank(userId));
+        checkArgument(isNotBlank(fileId));
+
+        ParticipantFile file = participantFileDao.getParticipantFile(userId, fileId)
+                .orElseThrow(() -> new EntityNotFoundException(ParticipantFile.class));
+        file.setDownloadUrl(getDownloadUrl(file));
+        return file;
+    }
+
+    /**
+     * Logs the metadata in the database, and then returns the passed-in ParticipantFile with pre-signed URL
+     * for S3 file upload. If the file or the file metadata already exists, throws EntityAlreadyExistsException.
+     *
+     * @param file the file metadata to be upload
+     * @return the ParticipantFile with pre-signed S3 URL for file upload.
+     * @throws org.sagebionetworks.bridge.exceptions.EntityAlreadyExistsException
+     *         if the file already exists.
+     */
+    public ParticipantFile createParticipantFile(ParticipantFile file) {
```

Comment:
I believe that the userID field is set by the Controller from the current active session, so the user cannot have access to other users' files...?

Reply:
We should move the logic from the Controller to the Service. This is because (a) we should put all the business logic in the Service, so that if we need to migrate Controllers, we can do so easily and (b) if another service or controller wants to call ParticipantFileService, the logic is already in the Service, so they don't need to replicate it.